src/include_graph_dependencies.cpp

```diff
@@ -5,6 +5,18 @@
 namespace clangmetatool {
 
 namespace {
+  
+// Backup copy of usage_reference_count, only used for internal implementation
+std::map<clangmetatool::types::FileGraphEdge, size_t> _usage_reference_count_bk;
+
+// Do copy for backup only once
+inline void backup_once(collectors::IncludeGraphData* data){
```

Comment:
There is a discussion about where we should call `backup_once`, we should always call it when we sure clangmetatool finished constructing the include graph.

So far there are two potential solution:
- call it in the end of frontend action(make sure after all AST matchers, ppcallbacks finished)
- call it in `IncludeGraphDependencies` functions that do write to `IncludeGraphData`

The trade-off is:
- first solution we only need to put `backup_once` once in the source while the second solution we need to make sure putting it in every functions in `IncludeGraphDependencies` that do write to `IncludeGraphData`(although so far only one function does the write, may add more write functions in the future)
- second solution is kind of copy on write strategy so looks more efficient so that only copy when it is needed, and logically easier to understand because when we start to write to `IncludeGraphData` in `IncludeGraphDependencies` the graph must be constructed already